packages/netlify-cms-core/src/actions/mediaLibrary.js

```diff
@@ -326,6 +360,27 @@ export function mediaPersisted(asset, opts = {}) {
   };
 }
 
+export function addMediaFilesToLibrary(mediaFiles) {
+  return (dispatch, getState) => {
+    const state = getState();
+    const action = {
+      type: ADD_MEDIA_FILES_TO_LIBRARY,
+      payload: { mediaFiles },
+    };
+    // add media files to library only after the library finished loading
+    if (state.mediaLibrary.get('isLoading') === false) {
```

Comment:
I can't believe I had to do this, but when loading a backup for a new unsaved entry directly from the new entry url (e.g. `/collections/posts/new`) the backup is loaded before the media library is mounted (hence loaded). Once the library did finish loading the entry media files were overwritten in the media library reducer.
Another solution was not to overwrite at all on media library load success, but it does makes sense to wait for the library to finish initial load before using it.

Reply:
Yeah this is fine for now at least. It's a code smell, but we have stuff like `isLoading` and `isFetching` for a reason.

Reply:
When you start syncing actions order inside `componentDidUpdate` based on props you know something is not right.

A better way to wait for actions is to use `redux-saga` https://redux-saga.js.org/docs/advanced/FutureActions.html as it uses a pull approach, but it might be an overkill for now.